### PR TITLE
Add mechanism for re-loading the statistic files in live

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
Statistic files will be changed by a supervisor script in live for future uses. This PR adds the ability to check the sha1 hashes of the file and compare it to a stored copy so that if the file has changed, it can be reloaded.

In this usage, the file in the command line arguments would be overwritten (or a symlink which is updated), and then the statistic would refresh at an appropriate rate.

## Standard information about the request

This is a new feature
This change affects the live search; Offline code is touched but not affected
This change changes scientific output

This change follows style guidelines (See e.g. [PEP8](https://peps.python.org/pep-0008/)), has been proposed using the [contribution guidelines](https://github.com/gwastro/pycbc/blob/master/CONTRIBUTING.md)

## Motivation
We want to be able to use a proper ranking statistic in Live, but do not want to have to use static files, or restart the search regularly in order to update them. This change means that the ranking statistic files can be reloaded

## Contents
 - Add functions to check the hashes of all files associated with the statistic object
 - Add threads for each statistic object which will monitor if the files have changed. These sleep until the next refresh time
 - Add functions to each statistic which detects whether the files needed by that statistic have been updated, and runs the appropriate function needed to reload the information from that file.

## Links to any issues or associated PRs
Link to draft PR: https://github.com/GarethCabournDavies/pycbc/pull/5/files
#4689, #4527, #4813

Note that this will significantly clash with #4608 

## Testing performed
 - Started pycbc live - printed some values from the fits files.
 - Updated fits files, checked that the values are updated after this, and that the updated files were logged

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: pycbc/events/stat.py
Comment (GarethCabournDavies) [reply pycbc/events/stat.py]: Thanks for pointing that out - for some reason I had it in my head that if you had if statements which were never met, then the else would be implemented. Kind of backwards to what it actually is I suppose.

I have updated to correct functionality with the latest commit

**Comment by GarethCabournDavies** [comment]
I think the failing tests are in a dependency, I'll ask in the code slack channel

**Review by maxtrevor — APPROVED** [review]
Approving now that Gareth has fixed the faulty code. CI is still blocking merge but I don't think its related to any of the changes made in this merge request.

**Comment by maxtrevor** [comment]
@GarethCabournDavies Looks like all of the required checks have now passed. Are you ready to merge?

**Comment by GarethCabournDavies** [comment]
Yup, I just hadn't noticed that the latest reattempt had worked!

**Pull request merged**
